src/main/python/hydra/zmqtest/zmq_pub.py

```diff
@@ -0,0 +1,67 @@
+__author__ = 'sushil, abdullahS'
+
+import zmq
+import random
+import time
+import logging
+import os
+from hydra.lib import util
+from hydra.lib.hdaemon import HDaemonRepSrv
+from hydra.zmqtest import zmq_config
+l = util.createlogger('HPub', logging.INFO)
+
+
+
+def send_pub_data(socket, msg_count=10000, msg_delay=0.01):
+    l.info("PUB server initiating sending [%d] messages, with delay[%f]", msg_count, float(msg_delay))
+    for index in range(msg_count):
+        messagedata = "msg%d" % index
+        #print "%d %s" % (index, messagedata)
+        #l.info("%d %s" % (index, messagedata))
+        socket.send("%d %s" % (index, messagedata))
+        time.sleep(float(msg_delay))
```

Comment:
sleep needs to be batched in a way that it triggeres every "n" packets. sleeping every packet is killing the performance